lxqt-session/src/sessionapplication.cpp

```diff
@@ -40,6 +43,14 @@ SessionApplication::SessionApplication(int& argc, char** argv) :
 {
     listenToUnixSignals({SIGINT, SIGTERM, SIGQUIT, SIGHUP});
 
+    {
+        // Checks wayland session
+        QString platform = QGuiApplication::platformName();
+	qDebug() << "Platform" << platform;
+	if(platform == QStringLiteral("wayland"))
+		qDebug() << "\tWayland session";
```

Comment:
I doubt it's necessary to add these.

Reply:
You are right.

I put it there to remember how to check de Wayland session. I don't like how Qt detects the platform. I would like something like:
`enum PLATFORM {`
`X11, WAYLAND, OTHER`
`};`

`Platform getPlaform() {`
`  String platform = QGuiApplication::platformName();`
`	if(platform == QStringLiteral("wayland"))`
`		return PLATFORM::WAYLAND;`
`        else if(platform == QStringLiteral("x11"))`
`                return PLATFORM::X11;`
`        return PLATFORM::OTHER;`
`}`

Maybe it should be included in lxqtlib. I realized that lxqt-config-appearance fails also.

Reply:
> I realized that lxqt-config-appearance fails also in Wayland platform.

Yes, some components make use of X11 without checking X11; some others, like qterminal and pcmanfm-qt, work fine under Wayland and X11 alike. We should fix the first group. This is a good start, IMO.

Reply:
Ok. I go for it.

Reply:
[https://github.com/lxqt/liblxqt/pull/286](https://github.com/lxqt/liblxqt/pull/286) is used to detects X11 or Wayland session.

Reply:
lxqt/liblxqt#286 doesn't detect the platform in  different way. It just wraps it.